internal/storage/v1/badger/spanstore/reader.go

```diff
@@ -444,6 +444,8 @@ func mergeJoinIds(left, right [][]byte) [][]byte {
 		case -1:
 			// left < right, increase left one
 			l++
+		default:
```

Comment:
Doesn't make sense in this context. I'd rather move case 0 to default (and change the others to greater/less 0)